jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/build/BuildConfiguration.java

```diff
@@ -228,6 +228,8 @@
   @Setter(AccessLevel.PRIVATE)
   private File dockerArchiveFile;
 
+  private List<String> cacheFrom;
```

Comment:
Given this is a list, to improve the builder, maybe we should make this a `@Singular` (You'll need to tune the name, because `cacheFrom` is already singular - not standard)

Reply:
Should I rename this to cacheFromList? In Docker REST API spec also has cachefrom name despite it being a JSON array:
![Screenshot_20201203_200959](https://user-images.githubusercontent.com/13834498/101043431-53b1c500-35a4-11eb-8d3b-926b98c81c34.png)

Reply:
No, you need to keep the name so it's compatible with the docs/mojo deserialization (I think).
If it's not necessary to keep the name, just rename to `caheFroms` with a final `s`. Lombok will generate methods `cacheFrom` to add entries to the list. Also keep in mind that when adding the singular annotation this value will be initialized as an empty list, so whenever we are using the getter, we need to consider the possibility of the list being empty.

Reply:
I think I can go ahead and rename the field to `cacheFroms`

Reply:
I really think that this gets populated from the XML provided by [<build>](https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin#config-image-build) element. So probably renaming will break any configuration that uses it. Double check, but I'm afraid name must remain intact.

You can configure `@Singular` to use a different method name `@Singular("addCacheFrom")` or `@Singular("cacheFromEntry")` might work well.